cpu/atmega_common/periph/timer.c

```diff
@@ -136,9 +137,35 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
         return -1;
     }
 
+    unsigned state = irq_disable();
     ctx[tim].dev->OCR[channel] = (uint16_t)value;
     *ctx[tim].flag &= ~(1 << (channel + OCF1A));
     *ctx[tim].mask |= (1 << (channel + OCIE1A));
+    irq_restore(state);
+
+    return 0;
+}
+
+int timer_set(tim_t tim, int channel, unsigned int timeout)
+{
+    if (channel >= TIMER_CHANNELS) {
+        return -1;
+    }
+
+    unsigned state = irq_disable();
+    unsigned absolute = ctx[tim].dev->CNT + timeout;
+    ctx[tim].dev->OCR[channel] = absolute;
+    uint8_t mask = 1 << (channel + OCIE1A);
+    *ctx[tim].mask |= mask;
+    if ((absolute - ctx[tim].dev->CNT) > timeout) {
+        /* Timer already expired. Make sure that interrupt is indeed triggered
+         */
+        while (!(*ctx[tim].flag & (1 << (OCF1A + channel)))) {
```

Comment:
```suggestion
        while (!(*ctx[tim].flag & mask)) {
```

Reply:
Note: `mask == (1 << (OCIE1A + channel)) != (1 << (OCF1A + channel))`

Reply:
One is for enabling, the other is the IRQ flag

Reply:
arg sorry, I should have looked closer.
Then that extra mask variable makes little sense